sdk/go/src/sawtooth_sdk/messaging/connection.go

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+// Package messaging handles lower-level communication between a transaction
+// processor and validator.
+package messaging
+
+import (
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	zmq "github.com/pebbe/zmq4"
+	uuid "github.com/satori/go.uuid"
+	"sawtooth_sdk/logging"
+	"sawtooth_sdk/protobuf/validator_pb2"
+)
+
+var logger *logging.Logger = logging.Get()
+
+// Generate a new UUID
+func GenerateId() string {
+	return fmt.Sprint(uuid.NewV4())
+}
+
+// DumpMsg serializes a validator message
+func DumpMsg(t validator_pb2.Message_MessageType, c []byte, corrId string) ([]byte, error) {
+	msg := &validator_pb2.Message{
+		MessageType:   t,
+		CorrelationId: corrId,
+		Content:       c,
+	}
+	return proto.Marshal(msg)
+}
+
+// LoadMsg deserializes a validator message
+func LoadMsg(data []byte) (msg *validator_pb2.Message, err error) {
+	msg = &validator_pb2.Message{}
+	err = proto.Unmarshal(data, msg)
```

Comment:
It is being returned using named return values: https://tour.golang.org/basics/7